### PR TITLE
Add ability to limit requests from using the Cache-Control header to skip cache

### PR DESCRIPTION
**Pull request by azrikahar**
## Description

Currently users can circumvent the API cache by sending `Cache-Control` request header containing `no-store`, and it is being used by the data studio as well.

This PR adds an environment variable to make it possible to lockdown the ability to skip cache, except for requests coming from the Data Studio.

Fixes ENG-490

## Type of Change

- [ ] Bugfix
- [x] Improvement
- [ ] New Feature
- [ ] Refactor / codestyle updates
- [ ] Other, please describe:

## Requirements Checklist

- [x] New / updated tests are included
- [x] All tests are passing locally
- [x] Performed a self-review of the submitted code

If adding a new feature:

- [x] Documentation was added/updated. PR: directus/docs#314

### CODE REVIEW

File: api/src/utils/should-skip-cache.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+/**
+ * Whether to skip caching for the current request
+ *
+ * @param req Express request object
+ */
+
+export function shouldSkipCache(req: Request): boolean {
+	const env = getEnv();
+
+	// always skip cache for requests coming from the data studio based on Referer header
```
Comment (paescuj) [inline api/src/utils/should-skip-cache.ts:14]:
```suggestion
	// Always skip cache for requests coming from the data studio based on Referer header
```

File: api/src/env.ts [hunk 2]
```diff
@@ -291,6 +293,9 @@ const typeMap: Record<string, string> = {
 	DB_PORT: 'number',
 
 	DB_EXCLUDE_TABLES: 'array',
+
+	CACHE_SKIP_ALLOWED: 'boolean',
+
```
Comment (paescuj) [inline api/src/env.ts:298]: Do we need this? I guess there are quite a lot of other boolean envs where we don't force this.

**Review by paescuj — APPROVED** [review]
Just a small question... Otherwise LGTM 👍

File: api/src/env.ts (hunk 2 above)
Comment (azrikahar) [reply api/src/env.ts:298]: It is definitely not necessary for this PR to work. This is more of an attempt at "continuing" the discussion in https://github.com/directus/directus/pull/17309#discussion_r1090772083 to start doing it more, especially for new env vars 😄 we can also opt out of this for now depending on how we'd want to tackle this env-type-map topic 👍

Comment (paescuj) [reply api/src/env.ts:298]: I see, in that case let's keep it like this 😃👍

**Review by freekrai — APPROVED** [review]
Tested with `CACHE_SKIP_ALLOWED` set to true, and when I pass `no-store` in the header of the request, I see `no-store` returned in cache control and when I set to false and pass the same, I get the default of `public, max-age=300, s-maxage=0` returned.

**Pull request merged**